_health-care/_js/components/form-elements/DateInput.jsx

```diff
@@ -14,57 +15,60 @@ class DateInput extends React.Component {
 
   handleChange() {
     const date = {
-      month: this.refs.month.value,
-      day: this.refs.day.value,
-      year: this.refs.year.value
+      month: Number(this._month.value),
+      day: Number(this._day.value),
+      year: Number(this._year.value)
     };
 
-    // Update UI based on validation.
-    if (!this.validate(date.day, date.month, date.year)) {
-      this.setState({ hasError: true });
-    } else {
-      this.setState({ hasError: false });
-    }
-
-    // Publish change up.
-    this.props.onUserInput(date);
-  }
-
-  validate(day, month, year) {
-    // Use the date class to see if the date parses back sanely as a
-    // validation check. Not sure this is a great idea...
-    const adjustedMonth = Number(month) - 1;  // JS Date object 0-indexes months. WTF.
-    const date = new Date(year, adjustedMonth, day);
-    return date.getDate() === Number(day) &&
-      date.getMonth() === adjustedMonth &&
-      date.getFullYear() === Number(year);
+    this.props.onValueChange(date);
   }
 
   render() {
-    const errorClass = this.state.hasError ? 'usa-input-error' : '';
+    const isValid = isValidDate(this.props.day, this.props.month, this.props.year);
     return (
-      <div className={errorClass}>
+      <div className={isValid ? undefined : 'usa-input-error'}>
         <div className="usa-date-of-birth">
           <div className="usa-datefield usa-form-group usa-form-group-month">
-            <label htmlFor={`${this.id}-month`}>Month</label>
-            <input id={`${this.id}-month`} aria-describedby="dobHint"
-                className="usa-form-control" pattern="0?[1-9]|1[012]"
-                type="number" min="1" max="12" value={this.props.date.month}
-                ref="month" onChange={this.handleChange}/>
+            <label
+                className={isValid ? undefined : 'usa-input-error-label'}
+                htmlFor={`${this.id}-month`}>
+              Month
+            </label>
+            <input
+                className="usa-form-control"
+                id={`${this.id}-month`}
+                max="12"
+                min="1"
+                pattern="0?[1-9]|1[012]"
+                ref={(c) => { this._month = c; }}
```

Comment:
Are we still using refs? Also, what exactly is this function doing? I'm a little confused